joblib/numpy_pickle.py

```diff
@@ -563,15 +580,19 @@ def dump(value, filename, compress=0, protocol=None, cache_size=None):
     return [filename]
 
 
-def _unpickle(fobj, filename="", mmap_mode=None):
+def _unpickle(fobj, filename="", mmap_mode=None,
+              ensure_native_byte_order=True):
     """Internal unpickling function."""
     # We are careful to open the file handle early and keep it open to
     # avoid race-conditions on renames.
     # That said, if data is stored in companion files, which can be
     # the case with the old persistence format, moving the directory
     # will create a race when joblib tries to access the companion
     # files.
```

Comment:
```suggestion
    # files.
    if ensure_native_byte_order == "auto":
        ensure_native_byte_order = mmap_mode is not None
    if ensure_native_byte_order and mmap_mode is not None:
        raise ValueError(
            "It is not possible to ensure native byte ordering with"
            f"mmap_mode={unpickler.mmap_mode}."
        ) 
```

(same remark about the reach-ability of this exception from the public API. But I have the feeling that it is saner to make it explicit in the code that the case `ensure_native_byte_order and mmap_mode is not None` is physically not achievable in general.